### PR TITLE
Add FXIOS-8473 [v125] Add AMP mobile provider

### PR DESCRIPTION
**Pull request by tiftran**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8473)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18795)

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by tiftran** [comment]
Tagging @adudenamedruby @OrlaM  and @linabutler  for review 🙂, this is a draft without a version since we're not sure whether or not which version this is going to go out in (might be an uplift?) and we don't have mobile suggestions in RS yet.

File: firefox-ios/firefox-ios-tests/Tests/ClientTests/SearchViewControllerTests.swift [hunk 1]
```diff
@@ -87,7 +87,7 @@ class SearchViewControllerTest: XCTestCase {
 
     func testFirefoxSuggestionReturnsNoSuggestions() async throws {
         FxNimbus.shared.features.firefoxSuggestFeature.with(initializer: { _, _ in
-          FirefoxSuggestFeature(availableSuggestionsTypes: [.amp: false, .wikipedia: false])
+            FirefoxSuggestFeature(availableSuggestionsTypes: [.amp: false, .ampMobile: false, .wikipedia: false])
```
Comment (adudenamedruby) [inline firefox-ios/firefox-ios-tests/Tests/ClientTests/SearchViewControllerTests.swift:90]: Not sure if this extra space is correct indentation or not? Let's double check. lol. Also, this line looks long enough that Swiftlint might scream. If you've ran Swiftlint and it's hasn't that's fine. If not, either let's do that, or just break up this array onto new-lines.

**Review by adudenamedruby — COMMENTED** [review]
LGTM otherwise. Not approving till out of draft and with a version.

**Review by linabutler — COMMENTED** [review]
Looks great! Your `available-suggestion-types` is already paying off 😁

File: firefox-ios/firefox-ios-tests/Tests/ClientTests/SearchViewControllerTests.swift (hunk 1 above)
Comment (tiftran) [reply firefox-ios/firefox-ios-tests/Tests/ClientTests/SearchViewControllerTests.swift]: sooo yes, i found that weird looking too, but that seems like what xcode wants when i try to break the line there? swift lint locally wasn't screaming at me, but i agree the line is kinda long

**Comment by tiftran** [comment]
Hi @adudenamedruby, this is ready for formal review now, also if we need to do an uplift, what's the process of it?
I found https://github.com/mozilla-mobile/firefox-ios/wiki/Release-Checklist#back-port, is this correct or is there some other document that I should be referencing?

**Comment by adudenamedruby** [comment]
Heya @tiftran. I'll take a look tomorrow first thing. WHat version does this need to be uplifted to?

**Comment by tiftran** [comment]
@adudenamedruby , we would like to have it for 124 if possible

**Comment by adudenamedruby** [comment]
@Mergifyio backport release/v124

**Review by adudenamedruby — APPROVED** [review]

**Comment by adudenamedruby** [comment]
BR is being weird, so I had to close & reopen

**Comment by adudenamedruby** [comment]
looks like swiftlint is failing.

**Comment by tiftran** [comment]
oh grrrr, why wasn't swiftlint angry with me locally.... i'll fix that

**Comment by tiftran** [comment]
okie doke, @adudenamedruby apologies, can we please try again 🙈

**Pull request merged**